astropy/io/fits/hdu/image.py

```diff
@@ -1198,6 +1224,8 @@ def _verify(self, option='warn'):
 class _IndexInfo:
     def __init__(self, indx, naxis):
         if _is_int(indx):
+            if indx < 0:  # support negative indexing
+                indx = indx + naxis
```

Comment:
Note: this small change enables `ImageHDU.section` to support negative indexes.  I added corresponding tests to `astropy/io/fits/tests/test_image.py` below.